### PR TITLE
added clear message of the data encoded by using float32array

### PR DESCRIPTION
**Pull request by MohitMaheshwari1711**
### Description

Missing encoding of the waveform. Because the parameter is a Float32Array, one can reasonably assume that the encoding is f32 (that is, PCM data amplitude in [-1:1]). However it would be better to confirm this in writing instead of leaving developers to their guess. There's also no mention of planar/non-planar encoding, which is necessary to do anything with the data retrieved (unless the signal has a single channel).

Added text - "The time-domain data is encoded in `f32` format with the amplitude of samples in [-1:1]" for the following urls for clarity :

- https://developer.mozilla.org/en-US/docs/Web/API/AnalyserNode/getFloatFrequencyData
- https://developer.mozilla.org/en-US/docs/Web/API/AnalyserNode/getFloatTimeDomainData

### Motivation

Better clarity for users going through the docs

### Related issues and pull requests

Issue :- #25571

### CODE REVIEW

**Comment by teoli2003** [comment]
@padenot Can you confirm the format is `f32` (and not something different like `f32-planar`, …

(There will be some additional work to be done afterwards, but let's wait for the answer here)

**Comment by padenot** [comment]
As noted, planar and interleaved arrangements are equivalent if there is only a single channel of audio, and here there's always a single channel of audio ([reference from the spec]):

> When the current time-domain data are computed, the input signal must be [down-mixed](https://webaudio.github.io/web-audio-api/#channel-up-mixing-and-down-mixing) to mono as if [channelCount](https://webaudio.github.io/web-audio-api/#dom-audionode-channelcount) is 1, [channelCountMode](https://webaudio.github.io/web-audio-api/#dom-audionode-channelcountmode) is "[max](https://webaudio.github.io/web-audio-api/#dom-channelcountmode-max)" and [channelInterpretation](https://webaudio.github.io/web-audio-api/#dom-audionode-channelinterpretation) is "[speakers](https://webaudio.github.io/web-audio-api/#dom-channelinterpretation-speakers)".

Nothing prevents sample to exceed `[-1.0, 1.0]` in the time domain version of the function. For the frequency domain, the numbers are not really limited, number can be -Infinity when the input is silent, or over 1.0 if the input has peaks that exceed `[-1.0, 1.0]`.

[reference from the spec]: https://webaudio.github.io/web-audio-api/#time-domain-down-mixing

**Comment by MohitMaheshwari1711** [comment]
Hey @teoli2003 

Do let me know if we need to make any further changes for the same

File: files/en-us/web/api/analysernode/getfloatfrequencydata/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ browser-compat: api.AnalyserNode.getFloatFrequencyData
 
 {{ APIRef("Web Audio API") }}
 
-The **`getFloatFrequencyData()`** method of the {{domxref("AnalyserNode")}} Interface copies the current frequency data into a {{jsxref("Float32Array")}} array passed into it.
+The **`getFloatFrequencyData()`** method of the {{domxref("AnalyserNode")}} Interface copies the current frequency data into a {{jsxref("Float32Array")}} (the time-domain data is encoded in f32 format with the amplitude of samples in [-1:1]) array passed into it.
```
Comment (teoli2003) [inline files/en-us/web/api/analysernode/getfloatfrequencydata/index.md:10]:
```suggestion
The **`getFloatFrequencyData()`** method of the {{domxref("AnalyserNode")}} Interface copies the current frequency data into a {{jsxref("Float32Array")}} array passed into it. Each array value is a _sample_, the magnitude of the signal at a particular time.
```

File: files/en-us/web/api/analysernode/getfloattimedomaindata/index.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ browser-compat: api.AnalyserNode.getFloatTimeDomainData
 
 {{ APIRef("Web Audio API") }}
 
-The **`getFloatTimeDomainData()`** method of the {{ domxref("AnalyserNode") }} Interface copies the current waveform, or time-domain, data into a {{jsxref("Float32Array")}} array passed into it.
+The **`getFloatTimeDomainData()`** method of the {{ domxref("AnalyserNode") }} Interface copies the current waveform, or time-domain, data into a {{jsxref("Float32Array")}} (the time-domain data is encoded in `f32` format with the amplitude of samples in [-1:1]) array passed into it.
```
Comment (teoli2003) [inline files/en-us/web/api/analysernode/getfloattimedomaindata/index.md:10]:
```suggestion
The **`getFloatTimeDomainData()`** method of the {{ domxref("AnalyserNode") }} Interface copies the current waveform, or time-domain, data into a {{jsxref("Float32Array")}} array passed into it. Each array value is a _sample_, the magnitude of the signal at a particular time.
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Yes: as there are no constrains about being inside [-1; 1], we shouldn't mention this. Also mentioning `f32` is not useful as the distinction between planar and interleaved is moot in the case of a single channel)

**Review by teoli2003 — APPROVED** [review]

**Comment by teoli2003** [comment]
Thanks Paul for the clarification!

**Comment by djee-ms** [comment]
Ok but then that change doesn't clarify much in my opinion once the range has been removed. Unless we explain what is mentioned above in comments that the samples are typically in [-1:1] range but can get beyond that, and/or link some spec, the added sentence doesn't tell me much as a developer. At the very least, since "samples" is highlighted, I assume this means that there's a formal definition to it somewhere, which might gives more hint, and so it would be welcome to link that definition here so that developers can easily come to the conclusion that data is _typically_ in [-1:1] range (the "classic" PCM float32 sample format found in most audio contexts).

**Pull request merged**